### PR TITLE
Fix NATS HA orphan-repair resurrection and improve consistency

### PR DESCRIPTION
**Pull request by dobesv**
Fixes a defect where the NATS worker's orphan-repair scan read raw log entries instead of the effective history, causing it to resurrect logically-retracted tool rounds (via EditEntries or Rewind).

Updates mid-turn tool injection to use a consistent (read-your-writes) NATS load, ensuring it captures client edits/retracts immediately preceding the injection. Improves observability by logging NATS read errors during continuation-turn drains instead of silently swallowing them.

Introduces apply_log_mutations_nats helper to DRY up u64 sequence conversion logic across nats_client_session, agent_loop, and daemon. Performs cosmetic cleanup of redundant u64->usize casts and log! style.

Expands test coverage for log mutations (skip paths, mid-round retracts) and adds a worker integration test verifying that retracted orphans are not repaired.

Follow-up to #891 / plan harnx-nats-ha.

## Summary by CodeRabbit

* **New Features**
  * Added NATS-aware mutation application for durable log “effective” state and switched read-your-writes session loading to an async consistent path.
* **Bug Fixes**
  * Improved edit/retract/rewind handling by applying mutations before history rendering, final-response extraction, and session reconstruction.
  * Added safer bounds-checked sequence conversions and fallback behavior to avoid incorrect or resurrected state.
  * Prevented retracted/orphan messages and tool calls from being injected, repaired, or resurfaced.
* **Tests**
  * Expanded NATS integration/regression coverage for retract visibility, rewind truncation, and orphan tool-call repair avoidance, plus additional edge cases around invalid edit ranges and malformed replacements.

### CODE REVIEW

File: crates/harnx-core/src/session_reconstruct.rs [hunk 1]
```diff
@@ -11,12 +11,38 @@ use crate::tool::ToolCall;
...
+                usize::try_from(*seq).expect("JetStream seq fits usize"),
+                entry.clone(),
+            )
+        })
+        .collect();
+    apply_log_mutations(&raw_with_seq)
+        .into_iter()
+        .map(|(seq, entry)| (u64::try_from(seq).expect("effective seq fits u64"), entry))
+        .collect()
+}
```
[the remark this answers is not in the document]
Comment (dobesv) [reply crates/harnx-core/src/session_reconstruct.rs:50]: Fixed in 39f93f6976c. `apply_log_mutations_nats` now returns `anyhow::Result` and `bail!`s on the `u64->usize` overflow instead of panicking. Result-returning callers (`load_or_repair_session`, daemon drain) propagate via `?`; the callers that legitimately return non-Result (`fold_new_user_messages_since`, `reconstruct_session_state`, `extract_final_response`, thin-client history render) log a warning and fall back to their existing safe state. The reverse `usize->u64` conversion stays `expect` with a SAFETY comment — effective seqs are a strict subset of input seqs (the resolver only deletes/replaces; replacements inherit the EditEntries seq), so it cannot overflow once the inputs validated.

File: crates/harnx-runtime/src/nats_client_session.rs [hunk 2]
```diff
@@ -328,9 +324,10 @@ impl ThinClientSession {
     /// The sequence number of the appended EditEntries entry.
     pub async fn retract_user_message(&self, seq: u64) -> Result<u64> {
         let log = NatsSessionLog::new(self.jetstream.clone(), self.session_id.clone());
+        let seq = usize::try_from(seq).expect("JetStream seq fits usize");
         let edit_entry = SessionLogEntry::EditEntries {
-            from: seq as usize,
-            to: seq as usize,
+            from: seq,
+            to: seq,
```
[the remark this answers is not in the document]
Comment (dobesv) [reply crates/harnx-runtime/src/nats_client_session.rs]: Fixed in 39f93f6976c. Both `edit_user_message` and `retract_user_message` now use `usize::try_from(seq).context("JetStream seq does not fit into usize")?` so a bad seq fails the request cleanly instead of aborting the thin client.

File: crates/harnx-runtime/src/nats_worker/agent_loop.rs [hunk 3]
```diff
@@ -105,7 +101,7 @@ pub(crate) fn build_mid_turn_injection_callback(
         let backend = backend.clone();
         let cursor = Arc::clone(&cursor);
         Box::pin(async move {
-            let tail = match backend.load_events_blocking() {
+            let tail = match backend.load_events_consistent_blocking() {
```
[the remark this answers is not in the document]
Comment (dobesv) [reply crates/harnx-runtime/src/nats_worker/agent_loop.rs]: Fixed in 39f93f6976c. Added `load_events_consistent_async` on the backend and the mid-turn injection callback now `.await`s it instead of the blocking read — the tool-round injection path no longer blocks the worker thread.

File: crates/harnx-runtime/tests/nats_worker.rs [hunk 4]
```diff
@@ -1000,6 +1002,348 @@ async fn after_seq_observer_advances_and_consistent_read_honors_it() -> Result<(
...
+            role: MessageRole::User,
+            content: harnx_core::message::MessageContent::Text("late message".to_string()),
+            timestamp: None,
+            fence_token: None,
+        })
+        .await?;
+    log.append_event_async(&SessionLogEntry::EditEntries {
+        from: retracted_seq as usize,
+        to: retracted_seq as usize,
+        replacements: vec![],
```
[the remark this answers is not in the document]
Comment (dobesv) [reply crates/harnx-runtime/tests/nats_worker.rs:1083]: Fixed in 39f93f6976c. The JetStream seq conversions in these test setups now use `usize::try_from(...).expect("JetStream seq fits usize")`.

File: crates/harnx-runtime/tests/nats_worker.rs [hunk 5]
```diff
@@ -1000,6 +1002,348 @@ async fn after_seq_observer_advances_and_consistent_read_honors_it() -> Result<(
...
+    })
+    .await?;
+
+    MID_ROUND_APPEND_DONE.notify_one();
+    tokio::time::sleep(Duration::from_secs(2)).await;
+
```
[the remark this answers is not in the document]
Comment (dobesv) [reply crates/harnx-runtime/tests/nats_worker.rs:1092]: Fixed in 39f93f6976c. The retracted-orphan negative test now uses a deterministic barrier: it waits until `nats_metrics.lease_acquisitions` increments (proving the worker claimed the session and reached `load_or_repair_session` / the orphan scan) and then until `active_sessions_per_worker` returns to 0, before asserting. I verified it is not vacuous: reverting the production fix (scanning the raw log) makes the test FAIL (a ToolResults is synthesized for the retracted call — left:1 right:0); with the fix it passes. The mid-round-retract test already gated on its `MID_ROUND_RELOAD_SEEN` barrier.

File: crates/harnx-runtime/tests/nats_worker.rs [hunk 6]
```diff
@@ -1000,6 +1002,348 @@ async fn after_seq_observer_advances_and_consistent_read_honors_it() -> Result<(
...
+        assistant_texts
+    );
+
+    let reconstructed = reconstruct_state(
+        &entries
+            .iter()
+            .map(|(_, entry)| entry.clone())
+            .collect::<Vec<_>>(),
+    );
+    assert_eq!(reconstructed.turn_status, TurnStatus::Idle);
```
[the remark this answers is not in the document]
Comment (dobesv) [reply crates/harnx-runtime/tests/nats_worker.rs]: Good catch — fixed in 39f93f6976c. The rewind test now uses `reconstruct_state_from_nats(&entries)` so `Rewind.after_seq` is evaluated against the JetStream sequence coordinate system instead of `reconstruct_state`'s 0-based re-enumeration.
